File: springboot-message-api/src/main/java/com/example/demo/controllers/MessageController.java
package com.example.demo.controllers;

import com.example.demo.entities.Message;
import com.example.demo.services.MessageService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@CrossOrigin("*")
@RequestMapping("/api/v1/message")
public class MessageController {

    @Autowired
    private MessageService messageService;

    @GetMapping
    public List<Message> getAllMessage(){
        return messageService.getAllMessages();
    }

    @GetMapping("/{id}")
    public Message getMessageById(@PathVariable int id){
        return messageService.getMessageById(id);
    }

    @PostMapping
    public String addMessage(@RequestBody Message msg){
        messageService.addMessage(msg);
        return "Add Success";
    }

    @PatchMapping("/{id}")
    public String updateMessage(@PathVariable int id, @RequestBody Message msg){
        messageService.updateMessage(id, msg);
        return "Update Success";
    }

    @DeleteMapping("/{id}")
    public String deleteMessage(@PathVariable int id){
        messageService.deleteMessage(id);
        return "Delete Success";
    }

}

File: express-message-api/app/routes/controllers/message.controller.js
import messageService from "../services/message.service.js";

const getAllMessage = async (req, res) => {
  const result = await messageService.getAll();
  res.send({ result });
};

const getMessageById = async (req, res) => {
  const { id } = req.params;
  const result = await messageService.getById(id);
  res.send({ result });
};

const addMessage = async (req, res) => {
  const result = await messageService.add(req.body);
  res.send({ result });
};

const updateMessage = async (req, res) => {
  const { id } = req.params;
  const result = await messageService.update(id, req.body);
  res.send({ result });
};

const deleteMessage = async (req, res) => {
  const { id } = req.params;
  const result = await messageService.deleteById(id);
  res.send({ result });
};

export {
  getAllMessage,
  getMessageById,
  addMessage,
  updateMessage,
  deleteMessage,
};
